README.md

```diff
@@ -106,19 +106,22 @@ Fetcher.registerService(myDataService);
 // register the middleware
 app.use('/myCustomAPIEndpoint', Fetcher.middleware());
 
-app.use(function(req, res, next) {
+app.use(function (req, res, next) {
     // instantiated fetcher with access to req object
     const fetcher = new Fetcher({
         xhrPath: '/myCustomAPIEndpoint', // xhrPath will be ignored on the serverside fetcher instantiation
-        req: req
+        req: req,
     });
 
     // perform read call to get data
     fetcher
         .read('data_service')
-        .params({id: ###})
-        .end(function (err, data, meta) {
-        // handle err and/or data returned from data fetcher in this callback
+        .params({ id: 42 })
```

Comment:
This is more of a style comment and less about the code changes, but I've never been a fan of this programming style. All of this code was written before async / await was available. I would prefer if the API were something simpler like:

```js
try { 
  // read(/* service */, /* params */, /* config */);
  const data = await fetcher.read('data_service', { id: 42 });
} catch (e) {
  console.log(e);
}
```

Maybe this is something we change for a 1.x release.

Reply:
Yup, I agree. I only fixed the docs because there are people at my company requiring a more up to date docs. I've also revisited what we discussed on #275 and this is what I want to tackle next. I also think it's possible to make it backward compatible so we can adjust everything before going to v1 and remove all the other APIs.

Reply:
Makes sense to me. BC is ideal but not a blocker if it makes the API better and user friendly.